From 9e0123871c43bfbfa273fa7b102dc72c63cf6b10 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 22 Feb 2019 13:55:12 +0100
Subject: [PATCH 1031/1051] phy: cp110-comphy: add support for 5G SFI/XFI
 comphy initialisation

The cp110-comphy supports 5G SFI/XFI modes on serdes lanes 2 and 4,
allow driver for triggering firmware to perform such configuration.

Change-Id: Ib5e6d3bc89130438d13cd1d26c4f1a9584d20cce
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/4404
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Marcin Wojtas <Marcin.Wojtas@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/phy/marvell/phy-mvebu-cp110-comphy.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
index 34f62f9af646..b9e8ea8a0bc6 100644
--- a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
+++ b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
@@ -128,6 +128,7 @@ static const struct mvebu_comhy_conf mvebu_comphy_cp110_modes[] = {
 	MVEBU_COMPHY_CONF_ETH(2, 0, PHY_INTERFACE_MODE_SGMII),
 	MVEBU_COMPHY_CONF_ETH(2, 0, PHY_INTERFACE_MODE_2500BASEX),
 	MVEBU_COMPHY_CONF_ETH(2, 0, PHY_INTERFACE_MODE_10GKR),
+	MVEBU_COMPHY_CONF_ETH(2, 0, PHY_INTERFACE_MODE_5GKR),
 	MVEBU_COMPHY_CONF(2, 0, PHY_MODE_PCIE),
 	MVEBU_COMPHY_CONF(2, 0, PHY_MODE_USB_HOST),
 	MVEBU_COMPHY_CONF(2, 0, PHY_MODE_SATA),
@@ -143,6 +144,8 @@ static const struct mvebu_comhy_conf mvebu_comphy_cp110_modes[] = {
 	MVEBU_COMPHY_CONF_ETH(4, 0, PHY_INTERFACE_MODE_SGMII),
 	MVEBU_COMPHY_CONF_ETH(4, 0, PHY_INTERFACE_MODE_2500BASEX),
 	MVEBU_COMPHY_CONF_ETH(4, 0, PHY_INTERFACE_MODE_10GKR),
+	MVEBU_COMPHY_CONF_ETH(4, 0, PHY_INTERFACE_MODE_5GKR),
+	MVEBU_COMPHY_CONF_ETH(4, 1, PHY_INTERFACE_MODE_5GKR),
 	MVEBU_COMPHY_CONF_ETH(4, 1, PHY_INTERFACE_MODE_SGMII),
 	MVEBU_COMPHY_CONF(4, 1, PHY_MODE_PCIE),
 	MVEBU_COMPHY_CONF(4, 1, PHY_MODE_USB_HOST),
@@ -236,6 +239,13 @@ static int mvebu_comphy_eth_power_on(struct phy *phy)
 						      lane->port,
 						      COMPHY_SPEED_3_125G));
 		break;
+	case PHY_INTERFACE_MODE_5GKR:
+		ret = data->comphy_smc(MV_SIP_COMPHY_POWER_ON, priv->phys,
+				 lane->id,
+				 COMPHY_FW_NET_FORMAT(COMPHY_XFI_MODE,
+						      lane->port,
+						      COMPHY_SPEED_5_15625G));
+		break;
 	case PHY_INTERFACE_MODE_10GKR:
 		ret = data->comphy_smc(MV_SIP_COMPHY_POWER_ON, priv->phys,
 				 lane->id,
-- 
2.17.1

